src/_P076_HLW8012.ino

```diff
@@ -165,25 +240,56 @@ boolean Plugin_076(byte function, struct EventStruct *event, String &string) {
     SaveCustomTaskSettings(event->TaskIndex, (byte *)&hlwMultipliers,
                            sizeof(hlwMultipliers));
 
-   Settings.TaskDevicePluginConfig[event->TaskIndex][4] = getFormItemInt(F("p076_current_read"));
-   Settings.TaskDevicePluginConfig[event->TaskIndex][5] = getFormItemInt(F("p076_cf_trigger"));
-   Settings.TaskDevicePluginConfig[event->TaskIndex][6] = getFormItemInt(F("p076_cf1_trigger"));
-
     if (PLUGIN_076_DEBUG) {
-      String log = F("HLW8012: Saved Calibration from Config Page");
-      addLog(LOG_LEVEL_INFO, log);
+     String log = F("HLW8012: Saved Calibration from Config Page");
+     addLog(LOG_LEVEL_INFO, log);
     }
 
     if (Plugin_076_hlw) {
-      Plugin_076_hlw->setCurrentMultiplier(hlwMultipliers[0]);
-      Plugin_076_hlw->setVoltageMultiplier(hlwMultipliers[1]);
-      Plugin_076_hlw->setPowerMultiplier(hlwMultipliers[2]);
+     Plugin_076_hlw->setCurrentMultiplier(hlwMultipliers[0]);
+     Plugin_076_hlw->setVoltageMultiplier(hlwMultipliers[1]);
+     Plugin_076_hlw->setPowerMultiplier(hlwMultipliers[2]);
     }
 
     if (PLUGIN_076_DEBUG) {
-      String log = F("HLW8012: Multipliers Reassigned");
+     String log = F("HLW8012: Multipliers Reassigned");
+     addLog(LOG_LEVEL_INFO, log);
+    }
+
+    //Set Pin settings
+    byte selectedDevice = getFormItemInt(F("p076_predefined_settings"));
+
+    Settings.TaskDevicePluginConfig[event->TaskIndex][7] = selectedDevice;
+
+    if (selectedDevice == 0){
+       Settings.TaskDevicePluginConfig[event->TaskIndex][4] = getFormItemInt(F("p076_current_read"));
+       Settings.TaskDevicePluginConfig[event->TaskIndex][5] = getFormItemInt(F("p076_cf_trigger"));
+       Settings.TaskDevicePluginConfig[event->TaskIndex][6] = getFormItemInt(F("p076_cf1_trigger"));
+    }
+    else {
+      if (selectedDevice < p076_PinSettingsCount){
+        Settings.TaskDevicePluginConfig[event->TaskIndex][4] = p076_PredefinedPinSettings[selectedDevice].Current_Read;
+        Settings.TaskDevicePluginConfig[event->TaskIndex][5] = p076_PredefinedPinSettings[selectedDevice].CF_Trigger;
+        Settings.TaskDevicePluginConfig[event->TaskIndex][6] = p076_PredefinedPinSettings[selectedDevice].CF1_Trigger;
+
+        //Not Working correct
+        //Settings.TaskDevicePin1[event->TaskIndex] = p076_PredefinedPinSettings[selectedDevice].SEL_Pin;
+        //Settings.TaskDevicePin2[event->TaskIndex] = p076_PredefinedPinSettings[selectedDevice].CF1_Pin;
+        //Settings.TaskDevicePin3[event->TaskIndex] = p076_PredefinedPinSettings[selectedDevice].CF_Pin;
+      }
+    }
+
+    if (PLUGIN_076_DEBUG) {
+      String log = F("HLW8012: PIN Settings ");
```

Comment:
The plugin will now support not only the HWL8012, so maybe just mention the plugin-ID in log sentences? (there are more of these)

Reply:
Fixed